src/pevm.rs

```diff
@@ -263,10 +266,16 @@ impl Pevm {
                         _ => unreachable!(),
                     }
                     // Assert that evaluated nonce is correct when address is caller.
-                    debug_assert!(
-                        tx.caller != address || tx.nonce.map_or(true, |n| n + 1 == info.nonce)
-                    );
-
+                    if tx.caller == address {
+                        if let Some(nonce) = tx.nonce {
```

Comment:
`Some(tx_nonce)` or `Some(expected_nonce)` is clearer.

Reply:
It should be `Some(observed_nonce)`, rather than `Some(expected_nonce)`.

To avoid ambiguity, let's just use `Some(tx_nonce)`.

Reply:
Yeah, let's go with `tx_nonce`.